### PR TITLE
feat: handle monetary number the right way

### PR DESCRIPTION
**Pull request by asbiin**
image

### CODE REVIEW

File: app/Helpers/MonetaryNumberHelper.php [hunk 1]
```diff
@@ -3,62 +3,168 @@
...
+        $value = static::formatValue($user, $amount, $currency);
+
+        if ($currency === null) {
+            return $value;
+        }
+
+        // // Other working example using MessageFormatter.
+        // $number = static::inputValue($amount, $currency);
+        // $base = \MessageFormatter::formatMessage(App::getLocale(), "{0, number}", [$number]);
+        // $formatted = \MessageFormatter::formatMessage(App::getLocale(), "{0, number, :: currency/$currency}", [$number]);
```
Comment (djaiss) [inline app/Helpers/MonetaryNumberHelper.php]: Should we keep those commented lines?

**Review by djaiss — COMMENTED** [review]
image

The first choice is always `locale` now.
Is that a bug?

**Comment by asbiin** [comment]
> image
> 
> The first choice is always `locale` now. Is that a bug?

Indeed, I just forgot to push a commit :D
I've fixed it, and updated some behavior.

File: app/Helpers/MonetaryNumberHelper.php (hunk 1 above)
Comment (asbiin) [reply app/Helpers/MonetaryNumberHelper.php]: Not necessarily. If needed, it's documented in this PR anyway...

**Comment by djaiss** [comment]
It's great! Let's go!

**Pull request merged**